From 3c5ca1e80bfd322ac03eba14d52af8f6e41fbdef Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 31 Mar 2018 22:46:57 +0200
Subject: [PATCH 1/2] Move mkchromecast.py to bin/mkchromecast, shebang fixes.

Distutils/setup.py runs into potential problems when there is both a
module mkchromecast and a script mkchromecast. Resolve issue by moving
script to new location bin/.

Drop the .py suffix since it is a script and make it 755.

Add sys.path code to make it run both when installed and git-cloned.

The shebang is problematic. /usr/bin/python is wrong on most platforms,
since this is a python3 script. Updating to /usr/bin/python3 which
should be fine in many but not all cases -  the upstream python packages
just installs /usr/bin/python3.6 etc., not python3.

When the shebang fails, invocations like "python3.6 scripts/mkchromecast"
will always work (and should be documented somewhere).

Packaging toolchains typically patches the shebang as required, so this
is less of a problem in downstream packaging.
---
 mkchromecast.py => bin/mkchromecast | 9 +++++++--
 setup.py                            | 1 +
 2 files changed, 8 insertions(+), 2 deletions(-)
 rename mkchromecast.py => bin/mkchromecast (98%)
 mode change 100644 => 100755

diff --git a/mkchromecast.py b/bin/mkchromecast
old mode 100644
new mode 100755
similarity index 98%
rename from mkchromecast.py
rename to bin/mkchromecast
index d687dd11..11f66818
--- a/mkchromecast.py
+++ b/bin/mkchromecast
@@ -1,6 +1,12 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3

 # This file is part of mkchromecast.
+import os.path
+import sys
+
+HERE = os.path.dirname(__file__)
+if os.path.exists(os.path.join(HERE, '..', 'mkchromecast')):
+    sys.path.insert(0, os.path.join(HERE, '..'))

 import mkchromecast.__init__
 from mkchromecast.version import __version__
@@ -10,7 +16,6 @@ import mkchromecast.colors as colors
 from mkchromecast.pulseaudio import create_sink, remove_sink
 from mkchromecast.utils import terminate
 import subprocess
-import os.path
 import time
 import atexit
 import signal
diff --git a/setup.py b/setup.py
index 4db4da0e..df47d6ac 100644
--- a/setup.py
+++ b/setup.py
@@ -32,6 +32,7 @@ DATA_FILES = [
         'images/google_working_w.icns',
         'images/google_nodev_w.icns',
         'bin/audiodevice',
+        'bin/mkchromecast',
         'nodejs',
         'notifier'
         ]
--
2.14.2

